libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/dialogs/QnAMakerDialog.java

```diff
@@ -620,21 +626,21 @@ public QnAMakerDialog(String sourceFilePath, Integer sourceLineNumber) {
         }
 
         if (dc.getContext().getActivity().getType() != ActivityTypes.MESSAGE) {
-            return EndOfTurn;
+            return CompletableFuture.completedFuture(END_OF_TURN);
         }
 
         QnAMakerDialogOptions dialogOptions = new QnAMakerDialogOptions() {
             {
-                setQnAMakerOptions(getQnAMakerOptions(dc));
-                setResponseOptions(getQnAResponseOptions(dc));
+                setQnAMakerOptions(QnAMakerDialog.this.getQnAMakerOptions(dc).join());
```

Comment:
This is weird `QnAMakerDialog.this`, is it necessary the `join()`?

Reply:
The `join()` is necessary cause the `setQnAMakerOptions` receives as a parameter an object (`QnADialogResponseOptions`) and not a CompletableFuture, so I need to do the Join(), same for the below line. Regarding the `QnAMakerDialog.this` if I don't do it, it references me to the same method in the `QnAMakerDialogOptions`, this [one](https://github.com/southworks/botbuilder-java/blob/feature/southworks/library/bot-ai-qna/base/libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/dialogs/QnAMakerDialogOptions.java#L22)

Reply:
But you can assign `this.getQnAMakerOptions(dc).join()` outside of the `new QnAMakerDialogOptions` and use the variable inside

Reply:
Yes, I can do that